index.ts

```diff
@@ -267,7 +268,7 @@ export default function generate(options: Options): Promise<void> {
 			}
 
 			// We can optionally output the main module if there's something to export.
-			if (options.main && options.main === (options.name + filenameToMid(sourceFile.fileName.slice(baseDir.length, -3)))) {
+			if (options.main && options.main === (options.name + filenameToMid(sourceFile.fileName.replace(fileExtRegExp, '').slice(baseDir.length)))) {	
```

Comment:
Let's remove the trailing whitespace please, which will retrigger the CI to see if this will pass now. Thanks!